ctapipe/core/container.py

```diff
@@ -108,6 +108,18 @@ def validate(self, value):
                 f"{errorstr} Should be an instance of {self.type}"
             )
 
+        if isinstance(value, Container):
+            # recursively check sub-containers
+            value.validate()
+
+        if isinstance(value, Map):
```

Comment:
shouldn't this return early? If value is a Map, the other validity checks don't need to run?

same above.

and if yes we may need to create a test that checks this?

Reply:
I added some returns for both cases. No need to check other attributes for Map or Container fields.